### PR TITLE
[ENH] Weibull distribution

### PR DESCRIPTION
**Pull request by malikrafsan**
#### Reference Issues/PRs

https://github.com/sktime/skpro/issues/22

#### What does this implement/fix? Explain your changes.

Weibull probability distribution

#### Does your contribution introduce a new dependency? If yes, which one?

no

#### What should a reviewer concentrate their feedback on?

- [ ] Energy formula for weibull distribution

#### Did you add any tests for the change?

- Yes, I try to install this library locally using `pip install --editable .[dev,test]` and create a simple driver program to use this new distribution

#### Any other comments?

* I use the formula from Wikipedia, which can be accessed [here](https://en.wikipedia.org/wiki/Weibull_distribution)

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/skpro/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/skpro/blob/main/.all-contributorsrc) in the `skpro` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks! Looks great!

* I'd name the parameters differently. The "lambda" or rate parameter is typically called the scale parameter, and identical with your current `scale` parameter. Your current `lambda_val` is the power or shape parameter, I would not call this "lambda" because that's typically the other parameter (your current `scale`).
* we should probably deal with negative `x`. Everthing is either 0 or na there.
* some further remarks on explicit forms below

File: skpro/distributions/weibull.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        """Probability density function."""
+        d = self.loc[x.index, x.columns]
+        # if x.values[i] < 0, then pdf_arr[i] = 0
+        pdf_arr = (d.lambda_val / d.scale) * (x.values / d.scale) ** (d.lambda_val - 1) * np.exp(- (x.values / d.scale) ** d.lambda_val)
+        return pd.DataFrame(pdf_arr, index=x.index, columns=x.columns)
+
+    def log_pdf(self, x):
+        """Logarithmic probability density function."""
+        d = self.loc[x.index, x.columns]
+        lpdf_arr = np.log(d.lambda_val / d.scale) + (d.lambda_val - 1) * np.log(x.values / d.scale) - (x.values / d.scale) ** d.lambda_val
```
[the remark this answers is not in the document]
Comment (malikrafsan) [reply skpro/distributions/weibull.py]: Is there any log that we can cancel? I can only think of storing division calculation

**Comment by malikrafsan** [comment]
> I'd name the parameters differently. The "lambda" or rate parameter is typically called the scale parameter, and identical with your current scale parameter. Your current lambda_val is the power or shape parameter, I would not call this "lambda" because that's typically the other parameter (your current scale).

Ahh yes, thank you so much for pointing this out!! I think previously I read the formula description reversed. I have adjusted this. Once again, thank you so much for pointing this out!

File: skpro/distributions/weibull.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (malikrafsan) [reply skpro/distributions/weibull.py]: Ahh I think I got it

[the remark this answers is not in the document]
Comment (fkiraly) [reply skpro/distributions/weibull.py]: I think the first option is numerically more stable, becaues over the course of the calculation your numbers stay in a narrower range.

**Comment by fkiraly** [comment]
Great! Let us know when you would like a review.

**Review by fkiraly — APPROVED** [review]
Thanks

**Pull request merged**